test/extended/images/append.go

```diff
@@ -0,0 +1,134 @@
+package images
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/image/dockerlayer"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func cliPodWithPullSecret(cli *exutil.CLI, shell string) *kapiv1.Pod {
+	err := exutil.WaitForServiceAccount(cli.KubeClient().Core().ServiceAccounts(cli.Namespace()), "builder")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	sa, err := cli.KubeClient().Core().ServiceAccounts(cli.Namespace()).Get("builder", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(sa.ImagePullSecrets).NotTo(o.BeEmpty())
+	pullSecretName := sa.ImagePullSecrets[0].Name
+
+	// best effort to get the format string for the release
+	router, err := cli.AdminAppsClient().Apps().DeploymentConfigs("default").Get("router", metav1.GetOptions{})
+	if err != nil {
+		g.Skip(fmt.Sprintf("Unable to find router in order to query format string: %v", err))
+	}
+	cliImage := strings.Replace(router.Spec.Template.Spec.Containers[0].Image, "haproxy-router", "cli", 1)
+
+	return &kapiv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "append-test",
+		},
+		Spec: kapiv1.PodSpec{
+			// so we have permission to push and pull to the registry
+			ServiceAccountName: "builder",
+			RestartPolicy:      kapiv1.RestartPolicyNever,
+			Containers: []kapiv1.Container{
+				{
+					Name:    "test",
+					Image:   cliImage,
+					Command: []string{"/bin/bash", "-c", "set -euo pipefail; " + shell},
+					Env: []kapiv1.EnvVar{
+						{
+							Name:  "HOME",
+							Value: "/secret",
+						},
+					},
+					VolumeMounts: []kapiv1.VolumeMount{
+						{
+							Name:      "pull-secret",
+							MountPath: "/secret/.dockercfg",
```

Comment:
`.docker/config.json`? (just to avoid confusion)  or is our SA secret really still in `.dockercfg` format?